.github/workflows/tests.yml

```diff
@@ -245,33 +245,33 @@ jobs:
       - name: Test (Node)
         env:
           RUSTFLAGS: -Dwarnings --cfg getrandom_backend="wasm_js"
-        run: wasm-pack test --node
+        run: wasm-pack test --node -- --features std
```

Comment:
Can we add `wasm-pack` test for the `wasm32v1-none` target?

Reply:
`wasm32v1-none` doesn't support `std`. I noted in the OP that I would need some guidance on how to add `no_std` tests to the project.

> I was unable to add actual `no_std` tests without some ugly modifications to it. Especially because of the whole `harness = false` thing. Let me know how you want me to proceed.

Reply:
Hm. I guess the best option will be to make code free of `std` items and since the code will be equivalent for both targets we can just rely on `wasm32-unknown-unknown` testing.

Reply:
The problem is specifically [disabling the default test harness](https://doc.rust-lang.org/1.83.0/cargo/reference/cargo-targets.html#the-harness-field). While `wasm-bindgen-test` is a test harness, it only works for Wasm. So if we disable the default test harness, native tests won't work.

I propose adding an explicit test target just for Web that is disabled by default:
```toml
[[test]]
name = "no_std-test"
path = "tests/mods.rs"
harness = false
test = false
```
Additionally we would have to add:
- `no_main`
- `no_std`
- A global allocator.

Which I propose could be added in a separate `web` module to not pollute the native tests.

WDYT?

Reply:
Wouldn't we have to copy the same tests to run in Wasm as well? Or are you proposing to create dedicated Wasm-only tests?

I guess you are saying we can do this in a follow-up or is that something you want done in this PR?

Reply:
Yes, we can leave addition of `wasm32v1-none` tests for a separate PR, but my main point is that we may not need them in the first place if there are no differences in code between `wasm32-unknown-unknown` and `wasm32v1-none`.

Reply:
Ah, I see!

Reply:
EDIT: I agree, if we remove the thread-local related code, we don't need to test that.